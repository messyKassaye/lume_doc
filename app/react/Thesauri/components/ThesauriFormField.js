import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { Field } from 'react-redux-form';
import { Icon } from 'UI';

export class ThesauriFormField extends Component {
  constructor(props) {
    super(props);
    this.focus = () => {
      this.groupInput.focus();
    };
  }

  renderValue(_value, index, groupIndex) {
    const { removeValue } = this.props;
    let model = `thesauri.data.values[${index}].label`;
    if (groupIndex !== undefined) {
      model = `thesauri.data.values[${groupIndex}].values[${index}].label`;
    }
    return (
      <div key={`item-${groupIndex || ''}${index}`}>
        <Field model={model}>
          <input className="form-control" type="text" placeholder="Item name" />
          <button
            tabIndex={index + 500}
            type="button"
            className="btn btn-xs btn-danger"
            onClick={removeValue.bind(null, index, groupIndex)}
          >
            <Icon icon="trash-alt" /> Delete
          </button>
        </Field>
      </div>
    );
  }

  render() {
    const { value, index, groupIndex } = this.props;
    return this.renderValue(value, index, groupIndex);
  }
}

ThesauriFormField.defaultProps = {
  groupIndex: undefined,
};

ThesauriFormField.propTypes = {
  removeValue: PropTypes.func.isRequired,
  value: PropTypes.shape({
    id: PropTypes.string,
    label: PropTypes.string,
  }).isRequired,
  index: PropTypes.number.isRequired,
  groupIndex: PropTypes.number,
};

export default ThesauriFormField;
