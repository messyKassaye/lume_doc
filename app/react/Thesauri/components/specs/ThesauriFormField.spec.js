import React from 'react';
import { shallow } from 'enzyme';
import { ThesauriFormField } from '../ThesauriFormField';

describe('ThesauriFormField', () => {
  let component;
  let props;
  beforeEach(() => {
    props = {
      value: {
        label: 'Item1',
        id: 'item1',
      },
      index: 1,
      removeValue: jest.fn(),
    };
  });

  const render = () => {
    component = shallow(<ThesauriFormField {...props} />);
  };

  it('should render thesaurus item field', () => {
    render();
    expect(component).toMatchSnapshot();
  });

  describe('delete button', () => {
    it('should remove item when clicked', () => {
      render();
      component
        .find('button')
        .first()
        .simulate('click');
      expect(props.removeValue).toHaveBeenCalledWith(props.index, undefined);
    });
    it('should pass groupIndex to removeValue if provided', () => {
      props.groupIndex = 5;
      render();
      component
        .find('button')
        .first()
        .simulate('click');
      expect(props.removeValue).toHaveBeenCalledWith(props.index, 5);
    });
  });
});
