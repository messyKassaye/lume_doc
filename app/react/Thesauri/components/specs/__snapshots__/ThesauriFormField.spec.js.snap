// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThesauriFormField should render thesaurus item field 1`] = `
<div
  key="item-1"
>
  <Modeled(Connect(Field))
    model="thesauri.data.values[1].label"
  >
    <input
      className="form-control"
      placeholder="Item name"
      type="text"
    />
    <button
      className="btn btn-xs btn-danger"
      onClick={[Function]}
      tabIndex={501}
      type="button"
    >
      <Connect(Icon)
        icon="trash-alt"
      />
       Delete
    </button>
  </Modeled(Connect(Field))>
</div>
`;
